rclrs/src/node/publisher.rs

```diff
@@ -35,7 +35,16 @@ impl Drop for PublisherHandle {
     }
 }
 
-/// Main class responsible for publishing data to ROS topics
+/// Struct for sending messages of type `T`.
+///
+/// Multiple publishers can be created for the same topic, in different nodes or the same node.
+///
+/// The underlying RMW will decide on the concrete delivery mechanism (network stack, shared
```

Comment:
Not entirely accurate, intraprocess may happen above the RMW layer

Reply:
I thought that is rclcpp-specific?

Reply:
Yes, that's what I meant, above the RMW layer. `rclcpp` implements intraprocess on top of RMW, whereas the documentation says:

> The underlying RMW will decide on the concrete delivery mechanism (network stack, shared memory, or **intraprocess**).

The RMW can't decide to use intraprocess.

Reply:
Maybe I'm wrong, but I'd assume that the RMW or DDS layer can detect when writer and reader are in the same process and use an optimized intraprocess delivery mechanism (independently from rclcpp having its own intraprocess mechanism). This is also how I'd read e.g. [this post](https://discourse.ros.org/t/ros2-foxy-rmw-fast-dds-improved-intra-process-inter-process-performance/17422).

Reply:
I thought you meant [`rclcpp`'s intra-process](https://docs.ros.org/en/galactic/Tutorials/Intra-Process-Communication.html), in that case, you're right about the docstring.